src/solvers/flattening/boolbv_equality.cpp

```diff
@@ -70,21 +72,25 @@ literalt boolbvt::convert_verilog_case_equality(
 
   if(!base_type_eq(expr.lhs().type(), expr.rhs().type(), ns))
   {
-    std::cout << "######### lhs: " << expr.lhs().pretty() << '\n';
-    std::cout << "######### rhs: " << expr.rhs().pretty() << '\n';
-    throw "verilog_case_equality without matching types";
+    error().source_location = expr.find_source_location();
+    error() << "verilog_case_equality without matching types:\n"
+            << "######### lhs: " << expr.lhs().pretty() << '\n'
+            << "######### rhs: " << expr.rhs().pretty() << eom;
+    throw 0;
   }
 
   const bvt &bv0=convert_bv(expr.lhs());
   const bvt &bv1=convert_bv(expr.rhs());
 
   if(bv0.size()!=bv1.size())
   {
-    std::cerr << "lhs: " << expr.lhs().pretty() << '\n';
-    std::cerr << "lhs size: " << bv0.size() << '\n';
-    std::cerr << "rhs: " << expr.rhs().pretty() << '\n';
-    std::cerr << "rhs size: " << bv1.size() << '\n';
-    throw "unexpected size mismatch on verilog_case_equality";
+    error().source_location = expr.find_source_location();
+    error() << "unexpected size mismatch on verilog_case_equality:\n"
+            << "lhs: " << expr.lhs().pretty() << '\n'
+            << "lhs size: " << bv0.size() << '\n'
+            << "rhs: " << expr.rhs().pretty() << '\n'
+            << "rhs size: " << bv1.size() << eom;
+    throw 0;
```

Comment:
CHECK_RETURN.

Reply:
Using DATA_INVARIANT because CHECK_RETURN doesn't allow me to add an error message.